src/basic/path-util.c

```diff
@@ -687,10 +696,20 @@ int find_executable_full(const char *name, bool use_path_envvar, char **ret_file
                 if (!path_is_absolute(element))
                         continue;
 
-                if (!path_extend(&element, name))
+                if (!path_extend(&path_name, name))
```

Comment:
use `element` here not path_name, that is the part of PATH that is being joined - ie, leave unchanged